app/controllers/application_controller.rb

```diff
@@ -91,6 +91,15 @@ def set_current_location_name
     cookies['location'] = params[:location] unless params[:location] == nil
   end
 
+  def check_for_empty_password
+    if session[:original_user] and session[:"warden.user.user.key"]
+      @user = User.find(session[:original_user])
```

Comment:
This shouldn't be @user=, but be just user=.  Using @ I believe will introduce the variable at a larger scope, which we wouldn't want in something as common as application_controller.